autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -867,225 +574,636 @@ def refit_single_full(self, X=None, y=None, X_val=None, y_val=None, models=None)
+        if unpersisted_models:
+            logger.log(20, f'Unpersisted {len(unpersisted_models)} models: {unpersisted_models}')
+        else:
+            logger.log(30, f'No valid persisted models were specified to be unpersisted, so no change in model persistence was performed.')
+        return unpersisted_models
+
+    def generate_weighted_ensemble(self, X, y, level, base_model_names, k_fold=0, n_repeats=1, stack_name=None, hyperparameters=None, time_limit=None, name_suffix='', save_bagged_folds=None, check_if_best=True, child_hyperparameters=None):
+        if save_bagged_folds is None:
+            save_bagged_folds = self.save_bagged_folds
+        if len(base_model_names) == 0:
+            logger.log(20, 'No base models to train on, skipping weighted ensemble...')
+            return []
+
+        # TODO: Remove extra_params, currently a hack
+        if child_hyperparameters is not None:
+            extra_params = {'_tmp_greedy_hyperparameters': child_hyperparameters}
+        else:
+            extra_params = {}
+        weighted_ensemble_model = WeightedEnsembleModel(
+            path=self.path,
+            name='weighted_ensemble' + name_suffix + '_k' + str(k_fold) + '_l' + str(level),
+            base_model_names=base_model_names,
+            base_model_paths_dict=self.get_models_attribute_dict(attribute='path', models=base_model_names),
+            base_model_types_dict=self.get_models_attribute_dict(attribute='type', models=base_model_names),
+            base_model_types_inner_dict=self.get_models_attribute_dict(attribute='type_inner', models=base_model_names),
+            base_model_performances_dict=self.get_models_attribute_dict(attribute='val_score', models=base_model_names),
+            hyperparameters=hyperparameters, eval_metric=self.eval_metric, stopping_metric=self.eval_metric, num_classes=self.num_classes, save_bagged_folds=save_bagged_folds, random_state=level + self.random_seed,
+            **extra_params
+        )
+
+        self._train_multi(X_train=X, y_train=y, X_val=None, y_val=None, models=[weighted_ensemble_model], k_fold=k_fold, n_repeats=n_repeats, hyperparameter_tune=False, feature_prune=False, stack_name=stack_name, level=level, time_limit=time_limit)
+        if check_if_best and weighted_ensemble_model.name in self.get_model_names_all():
+            if self.model_best is None:
+                self.model_best = weighted_ensemble_model.name
+            else:
+                best_score = self.get_model_attribute(self.model_best, 'val_score')
+                cur_score = self.get_model_attribute(weighted_ensemble_model.name, 'val_score')
+                if cur_score > best_score:
+                    # new best model
+                    self.model_best = weighted_ensemble_model.name
+        return [weighted_ensemble_model.name]
+
+    def _train_single(self, X_train, y_train, model: AbstractModel, X_val=None, y_val=None, **model_fit_kwargs) -> AbstractModel:
+        """Trains model"""
```

Comment:
maybe clarify: Trains model but does not add the trained model to this Trainer.

Reply:
added